liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateProcedureGenerator.java

```diff
@@ -90,11 +93,33 @@ public ValidationErrors validate(CreateProcedureStatement statement, Database da
             procedureText = procedureText.replace("OR REPLACE", "");
             procedureText = procedureText.replaceAll("[\\s]{2,}", " ");
         }
-        sql.add(new UnparsedSql(procedureText, statement.getEndDelimiter()));
+        if (database instanceof MSSQLDatabase) {
+            CreateProcedureGenerator.getSqlStatementsWithSet(sql, procedureText, statement);
+        } else {
+            sql.add(new UnparsedSql(procedureText, statement.getEndDelimiter()));
+        }
         surroundWithSchemaSets(sql, statement.getSchemaName(), database);
         return sql.toArray(new Sql[sql.size()]);
     }
 
+    public static void getSqlStatementsWithSet(List<Sql> sql, String procedureText, CreateProcedureStatement statement) {
+        String IS_SET_REGEX = "SET\\s(\\w)+\\s(ON|OFF);?";
```

Comment:
It will not handle SET statement like `SET     QUOTED_IDENTIFIER  ON `
Also, in the ticket mentioned that we need to leave only `QUOTED_IDENTIFIER` and `ANSI_NULL`
And this regexp will add all SET SOME_OPTION ON/OFF and it does not meet ticket requirements
MSSQL SET statements - https://docs.microsoft.com/en-us/sql/t-sql/statements/set-statements-transact-sql?view=sql-server-ver15

Reply:
@psrijan , I re-read my comment and realized that the comment is confusing.
I meant that `SET\\s(\\w)+\\s(ON|OFF);?` regexp will not handle multiple whitespace after `SET` keyword and after configuration property. Also, it does not handle lowercased and lowercased+uppercased statements.
I'm leaving for you different SQL statements to test regexp, make sure that regexp can find those statements.
- SET        QUOTED_IDENTIFIER ON;
- SET        QUOTED_IDENTIFIER                    ON;
- set QUOTED_IDENTIFIER              on;
- SET quoted_identifier ON;
- set quoted_identifier     OFF;

I recommend you to use a third-party tool to test regexp, for example, I use https://rubular.com/, but note that it's for ruby which means that once you implemented regexp, you will need to adapt it to java regexp format which as usually means adding an additional slash (`/`) to existing slash (e.g. `SET\s(\w)+...` is ruby format and java format looks like `SET\\s(\\w)+...`)
You can also try to find java specific tool to test regexp.

Reply:
done